src/app/features/public-form/public-form.component.ts

```diff
@@ -83,7 +83,19 @@ export class PublicFormComponent<E extends Entity> implements OnInit {
 
   private async loadFormConfig() {
     const id = this.route.snapshot.paramMap.get("id");
-    this.formConfig = await this.entityMapper.load(PublicFormConfig, id);
+
+    const publicForms = await this.entityMapper.loadType(PublicFormConfig);
+
+    const publicForm = publicForms.find(
+      (form: PublicFormConfig) => form.route === id,
+    );
+    const publicFormId = publicForm ? publicForm["_id"] : id;
+
+    this.formConfig = await this.entityMapper.load(
+      PublicFormConfig,
+      publicFormId,
+    );
+
```

Comment:
if you already found the form matching the route, you don't have to load it yet again from the DB. Just use the `publicForm` entity directly?

And event to match with the "_id", you could use an in-memory `publicForms.find(x => x.getId(true) === id)`, so we save one db request